migrations/20210612_initial.sql

```diff
@@ -43,6 +45,30 @@ CREATE TABLE contacts (
   UNIQUE (user_id, contact_profile_id)
 );
 
+CREATE TABLE sent_probes (
+  sent_probe_id INTEGER PRIMARY KEY,
+  contact_id INTEGER NOT NULL UNIQUE REFERENCES contacts ON DELETE CASCADE,
+  probe BLOB NOT NULL,
+  user_id INTEGER NOT NULL REFERENCES users,
+  UNIQUE (user_id, probe)
+);
+
+CREATE TABLE sent_probe_hashes (
+  sent_probe_hash_id INTEGER PRIMARY KEY,
+  sent_probe_id INTEGER NOT NULL REFERENCES sent_probes ON DELETE CASCADE,
+  contact_id INTEGER NOT NULL REFERENCES contacts ON DELETE CASCADE,
+  user_id INTEGER NOT NULL REFERENCES users,
+  UNIQUE (sent_probe_id, contact_id)
```

Comment:
would `UNIQUE (sent_probe_id, user_id)` be redundant?

Reply:
I think so, but we could make foreign keys include user_id as we spoke... It's a separate change though.